includes/class-wc-payments-order-success-page.php

```diff
@@ -73,6 +75,58 @@ public function show_woopay_payment_method_name( $payment_method_title, $abstrac
 		return ob_get_clean();
 	}
 
+	/**
+	 * Add the BNPL logo to the payment method name on the order received page.
+	 *
+	 * @param string            $payment_method_title the default payment method title.
+	 * @param WC_Abstract_Order $abstract_order the order being shown.
+	 */
+	public function show_bnpl_payment_method_name( $payment_method_title, $abstract_order ) {
+
+		// Only change the payment method title on the order received page.
+		if ( ! is_order_received_page() ) {
+			return $payment_method_title;
+		}
+
+		$order_id = $abstract_order->get_id();
+		$order    = wc_get_order( $order_id );
+		$bnpl_methods = [
+			'Affirm',
+			'Afterpay',
+			'Clearpay',
+			'Klarna',
+		];
+
+		if ( ! $order || ! in_array( $payment_method_title, $bnpl_methods, true ) ) {
+			return $payment_method_title;
+		}
+
+		$payment_method_id = $order->get_payment_method();
+		$payment_method    = WC()->payment_gateways()->payment_gateways()[ $payment_method_id ];
+
+		if ( ! $payment_method ) {
+			return $payment_method_title;
+		}
```

Comment:
```suggestion
		$payment_method_id = $order->get_payment_method();
		$payment_method = WC_Payments::get_payment_method_by_id( $payment_method_id )

		if ( ! $payment_method ) {
			return $payment_method_title;
		}
```
Couple points here. Do we need to restrict this functionality to BNPL payment methods? BNPL methods were mentioned as examples on the original issue, but I didn't read this as a restriction. Moreover, the rest of the code begins significantly simpler if we just remove this restriction and allow all payment method icons to be displayed, since you've already made the rest of this code dynamic.

Reply:
```suggestion
		$bnpl_methods = [
			'woocommerce_payments_' . Affirm_Payment_Method::PAYMENT_METHOD_STRIPE_ID,
			...
		];

		$payment_method_id = $order->get_payment_method();
		if ( ! $order || ! in_array( $payment_method_id, $bnpl_methods, true ) ) {
			return $payment_method_title;
		}

		$payment_method    = WC()->payment_gateways()->payment_gateways()[ $payment_method_id ];

		if ( ! $payment_method ) {
			return $payment_method_title;
		}
```
In any case, even if we are restricting the available payment methods here, we should be using the class constants to ensure this check on the title is not so brittle.